src/main/java/net/neoforged/neoforge/client/extensions/common/IClientItemExtensions.java

```diff
@@ -142,17 +145,22 @@ default Model getGenericArmorModel(ItemStack itemStack, EquipmentClientInfo.Laye
     default void setupModelAnimations(LivingEntity livingEntity, ItemStack itemStack, EquipmentSlot equipmentSlot, Model model, float limbSwing, float limbSwingAmount, float partialTick, float ageInTicks, float netHeadYaw, float headPitch) {}
 
     /**
-     * Called when the client starts rendering the HUD, and is wearing this item in the helmet slot.
+     * Called to render an overlay on the first-person camera.
      * <p>
-     * This is where pumpkins would render their overlay.
+     * This method will always be called to render an overlay, regardless of whether the client camera overlay provided
+     * by the {@link DataComponents#EQUIPPABLE Equippable data component} is present. If the equippable overlay is present
+     * (e.g. the pumpkin overlay), this method will be called after it has been rendered.
+     * <p>
+     * This method should be used if the overlay is dynamic or has dynamic components.
+     * For a static overlay, prefer using {@link Equippable#cameraOverlay()}.
      *
-     * @param stack       The item stack
-     * @param player      The player entity
-     * @param width       The viewport width
-     * @param height      Viewport height
-     * @param partialTick Partial tick time, useful for interpolation
+     * @param stack         The item stack that the player is wearing
+     * @param equipmentSlot The slot in which the player is wearing the above item stack
```

Comment:
```suggestion
     * @param equipmentSlot The slot in which the player is wearing or holding the above item stack
```

Since it runs for all equipment slots.